packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/TransformHelper.java

```diff
@@ -107,4 +109,87 @@ public static void processTransform(ReadableArray transforms, double[] result) {
       MatrixMathHelper.multiplyInto(result, result, helperMatrix);
     }
   }
+
+
+  private static double convertToDegrees(ReadableMap transformMap, String key) {
+    double value;
+    boolean inRadians = true;
+    if (transformMap.getType(key) == ReadableType.String) {
+      String stringValue = transformMap.getString(key);
+      if (stringValue.endsWith("deg")) {
+        inRadians = false;
+      }
+      if (stringValue.endsWith("rad") || stringValue.endsWith("deg")) {
+        stringValue = stringValue.substring(0, stringValue.length() - 3);
+      }
+      value = Float.parseFloat(stringValue);
+    } else {
+      value = transformMap.getDouble(key);
+    }
+    return inRadians ? MatrixMathHelper.radiansToDegrees(value) : value;
+  }
+
+
+  public static boolean tryProcessTransformBySkiaMatrix(ReadableArray transforms, Matrix matrix,
+                                                        int viewWidth, int viewHeight) {
+    assert matrix != null;
+    matrix.reset();
+    // center of view.
+    int originX = viewWidth / 2, originY = viewHeight / 2;
+
+    for (int transformIdx = 0, size = transforms.size(); transformIdx < size; transformIdx++) {
+      ReadableMap transform = transforms.getMap(transformIdx);
+      String transformType = transform.keySetIterator().nextKey();
+
+      if ("translate".equals(transformType)) {
+        ReadableArray value = transform.getArray(transformType);
+        double x = value.getDouble(0);
+        double y = value.getDouble(1);
+        double z = value.size() > 2 ? value.getDouble(2) : 0d;
+        // Not support translate in z axis.
+        if (z != 0) {
+          return false;
+        }
+        originX += x;
+        originY += y;
+        matrix.postTranslate(PixelUtil.toPixelFromDIP((float) x), PixelUtil.toPixelFromDIP((float) y));
+      } else if ("translateX".equals(transformType)) {
+        double x = transform.getDouble(transformType);
+        originX += x;
+        matrix.postTranslate(PixelUtil.toPixelFromDIP(x), 0);
+      } else if ("translateY".equals(transformType)) {
+        double y = transform.getDouble(transformType);
+        originY += y;
+        matrix.postTranslate(0, PixelUtil.toPixelFromDIP(y));
+      }
+    }
+
+    for (int transformIdx = 0, size = transforms.size(); transformIdx < size; transformIdx++) {
```

Comment:
You need to process all transforms in order, you can't have separate loops for translate and other properties, as it may cause the resulting matrix to be incorrect.

Reply:
I don't sure disorder  of  properties will leds a incorrect result matrix, since every property are handled by matrix multiply rather direct set matrix element.
 The reason for separate `translate` and other properties is we need to know the real center of the view to set origin of rotate and scale (default origin is the top-left point of the view, rather center of the view like it used to be)